xarray/core/groupby.py

```diff
@@ -567,10 +568,41 @@ def reduce(self, func, dim=None, axis=None, keep_attrs=False,
             Array with summarized data and the indicated dimension(s)
             removed.
         """
+        if dim == DEFAULT_DIMS:
+            dim = ALL_DIMS
+            # TODO change this to dim = self._group_dim after
+            # the deprecation process
+            if self._obj.ndim > 1:
+                warnings.warn(
+                    "Default reduction dimension will be changed to the "
+                    "grouped dimension after xarray 0.11. To silence this "
+                    "warning, pass dim=xarray.ALL_DIMS explicitly.",
+                    FutureWarning, stacklevel=2)
+        elif dim is None:
+            dim = self._group_dim
```

Comment:
I don't think we should add this yet -- this means that code that uses `groupby.reduce(... dim=None)` would change its behavior. There probably isn't much code like this but I don't see a good reason to change it yet.